staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/roundtripper_test.go

```diff
@@ -384,6 +403,258 @@ func TestRoundTripAndNewConnection(t *testing.T) {
+					shouldError:            true,
+				},
+				"proxied tcp->tcp via SOCKS5 with auth": {
+					proxyAuth:              url.UserPassword("proxyuser", "proxypasswd"),
+					serverConnectionHeader: "Upgrade",
+					serverStatusCode:       http.StatusSwitchingProtocols,
+					shouldError:            false,
+				},
+			}
+
+			for _, k := range sets.StringKeySet(testCases).List() {
+				testCase := testCases[k]
+				t.Run(k, func(t *testing.T) {
+					server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+						if testCase.shouldError {
+							if e, a := httpstream.HeaderUpgrade, req.Header.Get(httpstream.HeaderConnection); e != a {
+								t.Fatalf("%s: Expected connection=upgrade header, got '%s", k, a)
+							}
+
+							w.Header().Set(httpstream.HeaderConnection, testCase.serverConnectionHeader)
+							w.Header().Set(httpstream.HeaderUpgrade, "SPDY/3.1")
+							w.WriteHeader(testCase.serverStatusCode)
+
+							return
+						}
+
+						streamCh := make(chan httpstream.Stream)
+
+						responseUpgrader := NewResponseUpgrader()
+						spdyConn := responseUpgrader.UpgradeResponse(w, req, func(s httpstream.Stream, replySent <-chan struct{}) error {
+							streamCh <- s
+							return nil
+						})
+						if spdyConn == nil {
+							t.Fatalf("%s: unexpected nil spdyConn", k)
+						}
+						defer spdyConn.Close()
+
+						stream := <-streamCh
+						io.Copy(stream, stream)
+					}))
+					defer server.Close()
+
+					serverURL, err := url.Parse(server.URL)
+					if err != nil {
+						t.Fatalf("%s: Error creating request: %s", k, err)
+					}
+					req, err := http.NewRequest("GET", server.URL, nil)
+					if err != nil {
+						t.Fatalf("%s: Error creating request: %s", k, err)
+					}
+
+					spdyTransport := NewRoundTripper(nil, redirect, redirect)
+
+					var proxierCalled bool
+					var proxyCalledWithHost string
+					var proxyCalledWithAuth bool
+					var proxyCalledWithAuthHeader string
+
+					interceptor := &Interceptor{proxyCalledWithHost: &proxyCalledWithHost}
+
+					proxyHandler := socks5Server(nil, interceptor)
+
+					if testCase.proxyAuth != nil {
+						proxyHandler = socks5Server(&socks5.StaticCredentials{
+							"proxyuser": "proxypasswd",
+						}, interceptor)
+					}
+
+					closed := make(chan bool, 1)
+					isClosed := func() bool {
+						select {
+						case v := <-closed:
+							return v
+						default:
+							return false
+						}
+					}
+
+					l, err := net.Listen("tcp", "127.0.0.1:0")
+					if err != nil {
+						t.Fatalf("socks5Server: proxy_test: Listen: %v", err)
+					}
+					defer func() {
+						closed <- true
+						l.Close()
+					}()
+
+					go func() {
+						conn, err := l.Accept()
+						if err != nil {
+							if isClosed() {
+								return
+							}
+
+							t.Errorf("error accepting connection: %s", err)
+						}
+
+						if err := proxyHandler.ServeConn(conn); err != nil && !testCase.shouldError {
+							if isClosed() {
```

Comment:
nit: can you double check if we need to check `isClosed()` here? I think checking the l.Accept() error case is enough.

Reply:
Doesn't work without it, not sure why, leaving it for now.